pyvista/core/pointset.py

```diff
@@ -2047,6 +2048,42 @@ def cells(self, cells) -> None:
         vtk_idarr = numpy_to_idarr(cells, deep=False, return_ind=False)
         self.GetCells().ImportLegacyFormat(vtk_idarr)
 
+    @property
+    def faces(self) -> NumpyArray[int]:
+        """Return the polyhedron faces.
+
+        Returns
+        -------
+        numpy.ndarray
+            Array of faces.
+
+        """
+        if hasattr(self, 'GetPolyhedronFaces'):
+            faces = self.GetPolyhedronFaces()  # vtkCellArray
+            if faces is not None:
+                faces = faces.GetData()  # vtkCellArray -> vtkIdTypeArray
+        else:
+            faces = self.GetFaces()  # vtkCellArray
+        return convert_array(faces)
+
+    @property
+    def face_locations(self) -> NumpyArray[int]:
+        """Return polyhedron face locations.
+
+        Returns
+        -------
+        numpy.ndarray
+            Array of face locations.
+
+        """
+        if hasattr(self, 'GetPolyhedronFaceLocations'):
+            faces = self.GetPolyhedronFaceLocations()  # vtkCellArray
+            if faces is not None:
+                faces = faces.GetData()  # vtkCellArray -> vtkIdTypeArray
+        else:
+            faces = self.GetFaceLocations()  # vtkCellArray
+        return convert_array(faces)
```

Comment:
These return totally different results between <9.4 and 9.4+. Not sure how to handle this and its proving incompatible with the meshio conversion code from #7079

@jspanchu or @cquammen, are you familiar with the polyhedral cell storage layout changes and how we can correct the usage in the `to_meshio` conversion utility?

cc @keurfonluu

Reply:
the changes to storage are documented in https://docs.vtk.org/en/latest/release_details/9.4/vtk-polyhedral-cells-storage.html. Since it's only a deprecation, you could address it later by adapting the meshio conversion to handle new design.